### PR TITLE
Mobile responsive personal details page

### PR DESCRIPTION
**Pull request by onkar-saeloun**
Notion-
https://www.notion.so/saeloun/Mobile-Responsive-Personal-details-page-f9c32af89fde472dbd412ced3e908b8c?pvs=4

What-
- PR includes design elements of the mobile screen.
- updated in the custom input box, custom react select(common components) for mobile view
- updated routing for mobile view
- common components for mobile screen

Why-
- Common components need to behave responsive thus added `isDesktopView` flag
- HOC component needed to create and add on the component level instead of the wrapper over routing thus removed that

Note:-
1. This PR will need changes from the [API integration PR](https://github.com/saeloun/miru-web/pull/994) so can't be merged before that.
2. Verified the positive and negative scenarios same like API integration PR as both have same logic.

loom video-
https://www.loom.com/share/0739420b5fbb443198af7c233e45725b

### CODE REVIEW

**Review by prasanthchaduvula — APPROVED** [review]

File: app/javascript/src/components/Team/Details/PersonalDetails/MobilePersonalDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from "react";
+
+import dayjs from "dayjs";
+import { GlobeIcon, InfoIcon, MapPinIcon, PhoneIcon } from "miruIcons";
+
+import { Divider } from "common/Divider";
+import { InfoDescription } from "common/Mobile/InfoDescription";
+
+const MobilePersonalDetails = ({ personalDetails }) => (
+  <div className="mt-12 px-4 py-2">
```
[the remark this answers is not in the document]
Comment (onkar-saeloun) [reply app/javascript/src/components/Team/Details/PersonalDetails/MobilePersonalDetails.tsx:21]: Done

File: app/javascript/src/components/Team/Details/Layout/MobileNav.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+];
+
+const UserInformation = () => {
+  const {
+    details: { personalDetails },
+    updateDetails,
+  } = useTeamDetails();
+  const { memberId } = useParams();
+  const getDetails = async () => {
+    const res: any = await teamsApi.get(memberId);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/Details/Layout/MobileNav.tsx:57]: Move these to try-catch block and try to use promise life cycle methods if they are dependent API calls

File: app/javascript/src/components/Team/Details/Layout/MobileNav.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            </div>
+          </Tooltip>
+          <span className="text-xs leading-4 tracking-wider text-miru-dark-purple-1000" />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const TeamUrl = ({ urlList }) => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Team/Details/Layout/MobileNav.tsx:91]: Extract TeamUrl and UserInformation into components. Import them here and use them.

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Team/Details/Layout/MobileNav.tsx (hunk 2 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Team/Details/Layout/MobileNav.tsx]: done

File: app/javascript/src/components/Team/Details/Layout/MobileNav.tsx (hunk 3 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Team/Details/Layout/MobileNav.tsx]: done

**Comment by vipulnsward** [comment]
@onkar-saeloun Please rebase

**Review by prasanthchaduvula — APPROVED** [review]

File: app/javascript/src/context/UserContext.tsx [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable no-unused-vars */
```
Comment (vipulnsward) [inline app/javascript/src/context/UserContext.tsx:1]: Is this needed still?

**Review by vipulnsward — APPROVED** [review]

**Comment by vipulnsward** [comment]
@supriya3105  to UX review on Staging
@apoorv1316  for staging functionality review

**Pull request merged**